File: 804 - Unique Morse Code Words.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] code = new String[] {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> s = new HashSet<>();
        for (String word: words) {
            String trans = "";
            for (char i : word.toCharArray()) {
                trans += code[i - 97];
                // Character.code
            }
            s.add(trans);
        }
        return s.size();
    }
}

File: 804 - Unique Morse Code Words.cpp
#include <unordered_set>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        unordered_set<string> s;
        string code[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        for (string word : words) {
            string trans = "";
            for (char c : word) {
                trans += code[c - 97];
            }
            s.insert(trans);
        }
        return s.size();
    }
};